### DISCUSSION TITLE
How to dynamically route with query parameters

### DISCUSSION THREAD

**Discussion by initprism** [tokio-rs/axum#2185] [Q&A] [answered]
Unable to nest handler on same routing path in axum. Is there a way to solve this?

/path?id=abc
/path?id=abc&count=3
/path?page=1&count=3

**Comment by jplatte** [comment, marked as answer]
axum only does routing based on the path, and method. Query parameters should be handled with the [`Query`](https://docs.rs/axum/latest/axum/extract/struct.Query.html) extractor; you can internally make your handler defer the actual work to more fine-grained functions based on which parameters are set.
